### PR TITLE
Rule `no-test-arrow` disallows arrow functions in tests

### PR DESCRIPTION
**Pull request by azhang496**
ref: https://github.com/adobe-community/issues/issues/11412

### CODE REVIEW

**Comment by hzoo** [comment]
Maybe should add some tests for different number of arguments, different kinds of arguments?

Maybe do an early return for CallExpression on argument length > 2

**Comment by azhang496** [comment]
@hzoo are any of these functions called with more than 2 arguments tho? i thought `describe` and `it` would always be called with 2 args (string and function) and `beforeEach` etc. would be called with 1 function arg

**Comment by hzoo** [comment]
They aren't, but you can remove any checks for other CallExpressions if you want

**Comment by hzoo** [comment]
We should setup travis for this repo

**Comment by azhang496** [comment]
the condition would fail on `testFunctions.indexOf(node.callee.name) > -1` so the argument checking wouldn't even happen if it were another CallExpression

File: lib/rules/no-test-arrow.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+      });
+    }
+
+    return {
+
+      CallExpression(node) {
+        const testFunctions = ['describe', 'it', 'beforeEach', 'beforeAll', 'afterEach', 'afterAll'];
+
+        if (testFunctions.indexOf(node.callee.name) > -1 &&
```
Comment (mikesherov) [inline lib/rules/no-test-arrow.js]: Might be interesting....

`if (testFunctions.indexOf(node.callee.name) > -1 && node.arguments.some(arg => arg.type === 'ArrowFunctionExpression')) +{ `

File: README.md [hunk 2]
```diff
@@ -48,4 +48,6 @@ Then configure the rules you want to use under the rules section.
     * `imports`: deprecated libraries and their replacements
     * `expressions`: deprecated member expressions and their replacements
 
+* `no-test-arrow`: Disallows arrow functions as arguments to test functions (`describe`, `it`, `beforeEach`, `beforeAll`, `afterEach`, `afterAll`).
```
Comment (kaicataldo) [inline README.md]: I think we should specify this is for Jasmine test functions:
```
Disallows arrow functions as arguments to Jasmine test functions (`describe`, `it`, `beforeEach`, `beforeAll`, `afterEach`, `afterAll`).
```

**Comment by hzoo** [comment]
Travis is enabled now, so just got to add the .travis.yml file

**Comment by hzoo** [comment]
👍 nice I see the green checkmark too

File: lib/rules/no-test-arrow.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @fileoverview Rule that disallows passing arrow functions as arguments to karma/jasmine test functions
```
Comment (kaicataldo) [inline lib/rules/no-test-arrow.js]: `* @fileoverview Rule that disallows arrow functions as arguments to Jasmine test functions` for consistency

File: lib/rules/no-test-arrow.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+'use strict';
+
+// Rule Definition
```
Comment (kaicataldo) [inline lib/rules/no-test-arrow.js]: I think we can remove this comment

**Comment by kaicataldo** [comment]
A few small comments, but otherwise,

**Comment by mikesherov** [comment]
👍

**Pull request merged**